[reidenong] iP

reidenong opened the pull request:
# Barry the Chatbot

> "Hello! What tasks do you have to do today?" – Barry

Barry frees your mind of having to remember things you need to do. It's,

- text-based
- simple
- FAST **VERY FAST** to use

All you need to do is,

1. download it from [here](#).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

**Features:**

- [x] Managing tasks
- [x] Listing tasks
- [x] Listing different types of tasks
- [x] Finding tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

reidenong committed:
Level 0: Replaced Duke with Barry.

reidenong committed:
Level 1: Barry echoes user input.

reidenong committed:
Level 2: Add, List

reidenong committed:
Level 3: Mark, unmark as done

reidenong committed:
Level 4: Introduced different types of tasks

reidenong committed:
Level 5: Error Handling by Barry

reidenong committed:
Level 6: Delete tasks

reidenong committed:
Update README.md

reidenong committed:
Implemented Level 7: Save

reidenong committed:
Merge pull request #1 from reidenong/branch-Level-7

Implemented Level 7: Save

reidenong committed:
Level 8: Dates and Times

reidenong committed:
Merge pull request #2 from reidenong/branch-Level-8

Level 8: Dates and Times

reidenong committed:
A-MoreOOP: More OOP is used for barry

reidenong committed:
A-packages: Implemented barry package

reidenong committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

reidenong committed:
A-Gradle implemented

reidenong committed:
A-JUnit tests

reidenong committed:
A-JavaDoc

reidenong committed:
A-Find: Implemented the find function

reidenong committed:
Level 10 : Added GUI for Barry

reidenong committed:
Merge pull request #3 from reidenong/branch-Level-10

Level 10 : Added GUI for Barry

prave1n commented on src/main/java/barry/Parser.java:
```diff
@@ -0,0 +1,73 @@
+        String arguments = parts.length > 1 ? parts[1] : "";
+
+        switch (commandWord) {
+            case "bye":
```
there should be no indentation for case clauses

prave1n commented on src/main/java/barry/Task.java:
```diff
@@ -0,0 +1,128 @@
+ * This is an abstract base class for specific types of tasks, such as TodoTask, DeadlineTask, and EventTask.
+ */
+abstract class Task {
+    protected boolean completed;
```
boolean could be named isCompleted to follow the naming standards

prave1n commented on src/main/java/barry/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     * @return The task at the specified index.
+     * @throws BarryException If the index is out of the range of the task list.
+     */
+    public Task getTask(int index) throws BarryException {
```
as you already have a method named getTasks that returns an array of all the tasks, this method name could be confusing. perhaps you could rename it as getTaskAtIndex

prave1n reviewed (commented):
Looks good

SlothyCat commented on src/main/java/barry/Command.java:
```diff
@@ -0,0 +1,306 @@
+     * @throws BarryException If there is an error in the task operations.
+     * @throws IOException    If there is an error in input/output operations.
+     */
+    String execute(TaskList tasks, Ui ui, Storage storage) throws BarryException, IOException;
```
Very detailed comment. 😄  I understand that your classes have been well explained but will a small description for the other execute methods make them clearer? 🤔

SlothyCat commented on src/main/java/barry/Parser.java:
```diff
@@ -0,0 +1,73 @@
+                return new Command.FindCommand(arguments);
+            case "deadline":
+                String[] deadlineParts = arguments.split(" /by ");
+                if (deadlineParts.length < 2) throw new BarryException("The deadline format is incorrect.");
```
Exception is well handled. However, for the coding standard we are following in this course I believe we are supposed to put the conditional statement in another line. For example:

```java
if (condition) {
    action;
}

SlothyCat commented on src/main/java/barry/Storage.java:
```diff
@@ -0,0 +1,92 @@
+package barry;
+
+import java.io.*;
```
Small issue here, we have to explictly say what we have imported for this project. This might be an issue from the IDE settings. Could go settings > Editor > Code Style > Java > Imports 🤔

SlothyCat reviewed (commented):
LGTM overall very well done. Comments are very clear and intuitive. Could see the structure of the code really well. Just some mini issues that can be addressed quickly! 😆

reidenong committed:
Add README for Barry Task Manager

This commit introduces a detailed README file for Barry, the personal task manager chatbot. The README provides an overview of Barry's features and how it helps users manage tasks effectively.

Summary of the README:

- **Introduction**:
  - Brief description of Barry as a chatbot that helps users manage their tasks efficiently.

- **Features**:
  - Detailed explanation of Barry’s key features, including adding tasks, listing tasks, marking tasks as completed, deleting tasks, and finding specific tasks.

- **Task Management**:
  - Explanation of how Barry categorizes tasks into To-Dos, Deadlines, and Events.

- **User Commands**:
  - Overview of the user commands for managing tasks, including adding, marking, deleting, and searching tasks.

- **How Barry Helps**:
  - Summary of Barry's simplicity and how it helps users stay organized and productive.

---

This README will help users understand the capabilities of Barry and how to interact with it to manage their daily tasks.

reidenong committed:
Merge pull request #4 from reidenong/branch-A-JavaDoc

Add README for Barry Task Manager

reidenong committed:
Add Assertions to Document Important Assumptions in Parser Class

This commit introduces the use of Java's `assert` feature to document and enforce important assumptions throughout the `Parser` class. Assertions help ensure that the input and arguments passed to various commands meet expected conditions before processing, improving code robustness and preventing potential runtime errors.

Key Changes:
- Added assertions to ensure the following assumptions hold:
  - **Non-null and non-empty input**: Added assertions to check that user input is neither null nor empty before processing.
  - **Integer validation for commands**: Introduced assertions to validate that the arguments for commands such as `mark`, `unmark`, and `delete` are integers.
  - **Non-empty task descriptions**: For commands like `todo` and `find`, ensured that the task description or search term is not empty.
  - **Correct format for deadline and event commands**: Validated that the `deadline` command includes both the task and a valid `/by` date, and the `event` command contains both `/from` and `/to` times in the expected format.
  - **Valid date-time parsing**: Added assertions to confirm that the date and time strings are valid and in the correct format before attempting to parse them.

Purpose:
The assertions serve to document key assumptions in the code and provide early warnings if something goes wrong during runtime. This will make the application more reliable by detecting incorrect input formats or invalid data early in the execution process.

---

This change improves the overall code safety and prevents potential bugs caused by invalid or malformed input.

reidenong committed:
Improve Variable Naming in TaskList Class for Better Readability

This commit refactors the `TaskList` class by renaming several variables to improve clarity and code readability. The changes ensure that the names reflect their purpose more explicitly, making the code easier to understand without altering its functionality.

Key Changes:
- **Renamed `tasks` to `taskList`**: This makes it clear that the variable represents a collection of tasks.
- **Renamed `index` to `taskIndex`**: Provides more specific context, indicating that the index refers to the position of a task within the list.
- **Renamed `task` to `newTask`, `taskToMark`, `taskToUnmark`**: These new variable names clarify their specific role in the respective methods, making it obvious whether the task is being added, marked, or unmarked.

Purpose:
The renaming of variables enhances code readability and maintainability without affecting the behavior of the `TaskList` class. These changes are purely cosmetic and ensure that the codebase is easier to follow and less error-prone for future modifications.

No changes were made to the public API or behavior of the class, so existing functionality remains intact.

reidenong committed:
Add HelpCommand to Display Available Commands

This commit introduces a new `HelpCommand` to the Barry application. The `HelpCommand` provides users with a list of all available commands and their usage details. This helps users easily find out what commands they can run and how to structure them without needing to look elsewhere.

Key Changes:
- **New HelpCommand Class**: This command outputs a formatted list of all possible commands the user can execute, including `list`, `mark`, `unmark`, `todo`, `deadline`, `event`, `delete`, `find`, `bye`, and `help`.
- **Non-Exit Command**: The `HelpCommand` is not an exit command, so it keeps the application running after displaying the help message.
- **User Guidance**: Provides a clean and structured output to assist users in better understanding the available features of the Barry application.

This addition improves user experience by allowing them to see all supported commands at any time during the application’s runtime.

reidenong committed:
Created coding standard branch.

This is a backdated commit as i had implemented the coding standard changes in an earlier commit but forgot
to create a branch, so this branch is just so the git tracker for CS2103 can be updated.

reidenong committed:
Merge pull request #5 from reidenong/branch-A-CodingStandard

Created coding standard branch.

reidenong committed:
Added assert statements to document assumptions in TaskList class
A-Assertions

- Introduced assert statements at critical points in the TaskList class to enforce key assumptions about the program state.
  - Ensured task list is not null after initialization in both constructors.
  - Validated that new tasks added to the list are not null.

reidenong committed:
Enabled leave on bye and added cat image for barry mascot.

reidenong committed:
Renamed all Duke classes to match the new name of the chatbot, Barry

reidenong committed:
Updated the Barry User Guide, added the Ui image.

reidenong committed:
Improved README.md to include more user information.

reidenong committed:
Fixed missing directory bug in Storage class.

reidenong committed:
Added a Title for the Java GUI Window.

reidenong committed:
Generalized DiaglogBox method name

In the future, as one may wish to add on more chatbots,
the DialogBox object should not depend on the particular
chatbot but rather be ok to deal with all chatbots.

reidenong committed:
Added comments to Storage Object

The storage class was rather complicated and difficult
to understand, I added some comments to add to the clarity
of the code, facilitating future maintenance and upgrading.

reidenong committed:
Added comments to the UI class

The UI class lacked comments and this made it decently difficult
for one to see what was going on in the code. Some annotations
on the methods and what they meant would really help for future
maintenance or refactoring.

reidenong committed:
Added comments to the Main class

The Main class lacked comments and this made it decently difficult
for one to see what was going on in the code. Some annotations
on the methods and what they meant would really help for future
maintenance or refactoring.

reidenong committed:
Added comments to the MainWindow class

The MainWindow class lacked comments and this made it decently difficult
for one to see what was going on in the code. Some annotations
on the methods and what they meant would really help for future
maintenance or refactoring.

reidenong committed:
Add comments to barry class

Improvements will help future developers or maintainers
understand the structure and functionality of the
application more easily.

reidenong committed:
Improve comments for ui class

Updated comments to provide clearer explanations for each method,
including their purpose and return values. Refined class description
to better convey the Ui class' role in handling user interaction
through the command-line interface. Ensured consistent terminology
across all comments to improve readability and maintainability.

reidenong committed:
Improve comments in Storage class

Refined the comments within the Storage class to improve clarity and provide
detailed explanations of its methods and functionality.

reidenong committed:
Add comments to DialogBox class

Added detailed comments to the DialogBox class to improve
readability and understanding of the code

reidenong committed:
Add comments to MainWindow class

Added detailed comments to the MainWindow class to
clarify its structure and functionality

reidenong committed:
Add comments to the Launcher class

Added comments clarify the purpose of the Launcher class
